File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
   
    int k = 1;
    int officer = 0;
    int cm = 1;

    int length = nums.size();

    for(int i = 0; i < length - 1; i++) {

        if(nums[cm] == nums[cm-1])
            cm++;
        else{
            nums[officer+1] = nums[cm];
            k++;
            officer++;
            cm++;
        }
    }



    for(int i = 0; i < k; i++){
        cout << nums[i] << " ";
    }
    return k;
    }
};

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }
        if(nums.length==1)
        {
            return 1;
        }
        int k = 1;
        int cm = 1;
        int length = nums.length;
        for (int i = 0; i < length - 1; i++) 
        {
            if (nums[cm] == nums[cm - 1]) {
                cm++;
            } else {
                nums[k] = nums[cm];
                k++;
                cm++;
            }
        }
        
        return k;
    }
}